controllers/managedocs_controller.go

```diff
@@ -1482,3 +1524,18 @@ func (r *ManagedOCSReconciler) unrestrictedGet(obj client.Object) error {
 	key := client.ObjectKeyFromObject(obj)
 	return r.UnrestrictedClient.Get(r.ctx, key, obj)
 }
+
+func (r *ManagedOCSReconciler) ensureMCGDeployment(storageCluster *ocsv1.StorageCluster, enableMCGAsString string) error {
+	// Check and enable MCG in Storage Cluster spec
+	mcgEnable, err := strconv.ParseBool(enableMCGAsString)
+	if err != nil {
+		return fmt.Errorf("Invalid Enable MCG value: %v", enableMCGAsString)
+	}
+	if mcgEnable {
+		r.Log.Info("Enabling Multi Cloud Gateway")
+		storageCluster.Spec.MultiCloudGateway.ReconcileStrategy = "manage"
+	} else if storageCluster.Spec.MultiCloudGateway.ReconcileStrategy == "manage" {
+		r.Log.V(-1).Info("Trying to disable Multi Cloud Gateway, Invalid operation")
+	}
+	return nil
+}
```

Comment:
If this is a refactor of the MCG reconciliation then we should use it in both cases, converged and consumer.
Right now you are just using it in consumer. Can you please refactor converged to use it as well?